### PR TITLE
Update tests for increased code coverage

### PR DESCRIPTION
**Pull request by MichaelSquires**
- Updated `synapse.lib.cache.FixedCache.pop()` to remove the key from the fifo and the cache
- Updated `synapse.tests.test_lib_cache.py::CacheTest` to cover missing edge case in `synapse.lib.cache`
- Added test for `synapse.lib.reflect.getItemLocals()`
- Added test for `synapse.lib.reflect.getMethName()`
- Updated test for `synapse.tools.cryo.cat`:
  - Covered edge case for `--verbose`
  - Covered edge case for output when specifying `--msgpack`
- Created new test for `synapse.tools.cryo.list`
- Added check for synapse version check in:
  - `csvtool`
  - `hive.load`
  - `hive.save`
- Added `return 0` in `main()` of `cryo.list` and `hive.load` for consistent return values from tools

### CODE REVIEW

File: synapse/lib/cache.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ def __len__(self):
         return len(self.cache)
 
     def pop(self, key):
+        if key in self.fifo:
```
Comment (invisig0th) [inline synapse/lib/cache.py:30]: Is it possible that this is not getting tested? Seems like the relative coverage drop may be just here.

Comment (invisig0th) [reply synapse/lib/cache.py:30]: I think we originally "decided" not to delete from the fifo due to O(N) iteration concerns, but in retrospect that was probably dangerous due to potential GC side effects.  If we made the `self.cache.pop()` default to `s_common.novalu` and then only did the `self.fifo.remove()` in the event that it is not `novalu` we could probably split the difference. ( of course, remembering to return None if it was `novalu` to maintain backward compat )

Blind/unconfirmed `pop()` use cases should be minimal, but we can make an easy optimization.

Comment (MichaelSquires) [reply synapse/lib/cache.py]: Good call. I did some performance testing and `if key in self.fifo` was an order of magnitude slower. The implementation that you suggested (that just got committed) is comparable to the original implementation.
